### DISCUSSION TITLE
Simple figure cross reference does not work

### DISCUSSION THREAD

**Discussion by steven-wooding** [quarto-dev/quarto-cli#5111] [Q&A] [answered]
I'm trying to get a simple figure cross reference to work:

````
---
title: "Testing cross referencing"
---

Some test type @fig-airquality

```{r}
#| label: fig-airquality

a <- c(1, 2, 3)
a
```
````

(The code does NOT have the trailing bracket but I can't work out how to get back ticks to render without it)

This fails with 'WARNING: Unable to resolve crossref @fig-airquality'

I'm sure I'm doing something really silly wrong, but I can't see what it is.

I'm running RStudio 2023.03.0+386 on macOS arm64 13.2.1 and Quarto version 1.3.310

Any suggestions much appreciated.

**Comment by dragonstyle** [comment]
Try adding a `fig-cap` to the code chunk- figures require captions...

Hope this fixes it!

**Reply by steven-wooding** [reply]
Thanks for the super speedy reply - I'm afraid it doesn't fix it. I now have:
```
---
title: "Testing cross referencing"
---

Some test type @fig-airquality

```{r}
#| label: fig-airquality
#| fig-cap: "Air quality"

a <- c(1, 2, 3)
a
```{
```
And still get the error "WARNING: Unable to resolve crossref @fig-airquality"
and in the output the cross reference is rendered as "Some test type ?@fig-airquality"

**Reply by rgaiacs** [reply, marked as answer]
Can you use some code that produces an image? For example,

``````
```{r}
plot(c(1, 2, 3), c(1, 2, 1))
```
``````

**Reply by dragonstyle** [reply]
That's a good catch -  I think emitting a plot would fix it. To make a figure out of a non-plot, you can use figure divs, like:

````markdown
:::{#fig-airquality2}
```{r}
a <- c(1, 2, 3)
a
```

A set of data about air quality
:::
````

**Reply by steven-wooding** [reply]
It looks like the 'producing a graphic' is the key thing. The figure caption (#| fig_cap:) is not necessary to make it work.

Big thanks for helping me solve this.

I'm trying to troubleshoot a failure to cross-reference in a more complicated document and it looks like in making a minimal example I've actually changed the problem :-(. I'll try and come up with  a better example for the problem that started this.

**Comment by cscheid** [comment]
To respond to your code-formatting question: the way to get triple-ticks to work inside a code block is to use _more_ ticks on the outside code block than on the inside. So, use four ticks to open a code block that includes triple ticks, etc.

**Reply by steven-wooding** [reply]
Thanks - much appreciated. I searched and found the instructions to use 'more back ticks' but I couldn't work out _how_ to use the extra ticks!

**Comment by Prokophs** [comment]
To all the people strangling with quarto cross-references.
Below is a workaround that works while we wait for quarto to make changes regarding this issue, especially when working with word documents.

Do not use captions at this point.
For tables use the standard cross-reference structure using #tbl-yourcaption --> [@tbl-yourcaption].
For images, insert the image in a table and use the same structure above.

Render the document and when word appears do the following:
1. click on the crosshair to select the table, right click to open the menu and click on "insert caption" and a popup will show
2. type your caption, from options select the label, whether it is a table or a figure
3. choose where you want the caption placed, above or below
4. click ok

Return to the document and manually modify your cross-reference link inside the [ ] as a table or figure (for example [Figure 1] or [Table 1]).
To use list of tables or list of figures do the following:
1. click the "Reference" tab --> click "Insert table of figures" and a popup will show
2. choose whether you want to insert a list of tables or a list of figures

Hope it helps.

<img width="657" alt="SCR-20240316-jnyv" src="https://github.com/quarto-dev/quarto-cli/assets/6112600/ea87c6f8-4a34-4803-ad6a-b88c3e9d6353">
<img width="646" alt="SCR-20240316-jmxm" src="https://github.com/quarto-dev/quarto-cli/assets/6112600/384be12c-6ae4-451d-9cd2-20422ed24e00">

**Comment by mcanouil** [comment]
Which version of Quarto are you using?

Note that this thread is a year old and that the cross-reference system had a huge makeover in 1.4.

**Reply by Prokophs** [reply]
I'm using the latest version 1.4 and so far cross-references hasn't worked for word documents.
All major quarto updates are mostly for html and pdf outputs.

**Reply by mcanouil** [reply]
Anyway, this thread has nothing to do with Word.
In would be better, to open a new discussion and show with a reproducible example what is not working as you expect.

**Reply by mcanouil** [reply]
Note that the new cross-reference allows this to work:

````qmd
---
title: "Testing cross referencing"
---

Some test type @fig-airquality

::: {#fig-airquality}

```{r}
a <- c(1, 2, 3)
a
```

:::
````

**Reply by Prokophs** [reply]
Dear mcanouil, i'm just trying to help and not to have a discussion on this matter. 
There are tons of posts from people having issues when working with the word format.
Is list of tables or list of figures supported for word documents, or a placement of a caption above an image?
On the quarto website there is still nothing and the word limitations are known.

If you feel this post has nothing to do with this issue, please feel free to delete it.

**Reply by mcanouil** [reply]
@Prokophs The issue is has anything to do with your message being useful or not. This thread is unrelated to Word or List of tables/figures. You are simply off topic which means you bring confusion and making unlikely for future users to find this thread useful or what you posted.
And no, we don't delete content.

Please open a new discussion with what you bring, this way, this can be discussed, answered and this will be more discoverable for users.

Thank you for your cooperation.
